ingest/rules/curate.smk

```diff
@@ -66,9 +65,6 @@ rule curate:
             | augur curate transform-strain-name \
                 --strain-regex {params.strain_regex} \
                 --backup-fields {params.strain_backup_fields} \
```

Comment:
_non-blocking_

I'm wondering if this is no longer needed since there's special handling of the strain name with custom scripts. All this is really doing now is coping `isolate` to `strain` if the `strain` from Entrez is empty.

Reply:
Just noting that we discussed this offline. We’ll leave it as-is for now but can revisit it in the future.

Pros:
* Could reduce a step during the curation process.
* Clearly separates the "strain" and "isolate" columns to align with the GenBank record.

Cons:
* Since the Mumps strain name could be in either column, we would need to refactor the code to infer `genotype` and `district` from both the `strain` and `isolate` names (instead of just `strain`).
* In the live builds, which automatically display the strain name, the user would need to switch to displaying the isolate column to view the Mumps strain name.